pymodbus/client/udp.py

```diff
@@ -158,7 +158,7 @@ def protocol_lost_connection(self, protocol):
 
             self.connected = False
             self.protocol = None
-            if self.params.host:
```

Comment:
I don't have any UDP server, so haven't tried it

Reply:
I think it is covered by our test suite.

Reply:
and just for information, pymodbus have an excellent server, which are very good for testing cases like this.